backend/core/llm_chain.py

```
@@ -3,490 +3,317 @@
+    query_lower = query.lower()
+    selected_names = set()
 
-        logger.debug(f"Cache cleanup: {len(_response_cache)} entries, ~{total_size} bytes")
+    # Keyword-based routing logic
+    resume_keywords = ["experience", "job", "work", "skill", "resume", "cv", "company", "role", "hillman", "wisnet", "history"]
+    about_keywords = ["about", "background", "who is", "philosophy", "approach"]
+    illustration_keywords = ["art", "illustration", "drawing", "picture", "character", "design"]
 
-    except Exception as e:
-        logger.error(f"Error during cache cleanup: {e}")
+    if any(keyword in query_lower for keyword in resume_keywords): selected_names.add("resume")
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)